src/main/java/com/rybka/service/ApiConnectorService.java

```diff
@@ -0,0 +1,31 @@
+package com.rybka.service;
+
+import com.rybka.exception.NullResponseException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class ApiConnectorService {
+
+    private static final String API_KEY = "4e0bbdc44fcdf05612fa0882";
+
+    public HttpResponse<String> retrieveRates(String userBaseCurrency) throws NullResponseException {
```

Comment:
Why do you return HttpResponse here? Return List<ExchangeRate>

Reply:
fixed, but return Map<String, Double>